### PR TITLE
fix: install params section on marketplace deploy

### PR DESCRIPTION
**Pull request by polillomm**
a) show the section always open;
b) put "required" into the fields that are required so the form can't be submitted empty;
c) auto-generate password and let it visible so people can copy it;
d) fill all the fields with dummy values, say adminUsername, adminMail etc, so people have to clear the field and fill with their own info but have an example of exactly what a valid value consists => on Ez we use Star Trek characters names and fake mails for example;
e) Move the marketplace assets to the repository, not just the images, and adjust the recipes for this;
f) Adjust the marketplace and services recipes to use the files we already have via git clone;
g) Adjust the marketplace and services installation code to include the system data field that stores the assets directory path so that the substitutions in the cmd steps role with this data field work as well;
h) Restart NGINX when SSL is issued;
i) Update the SRI regex to match Ez's version;
j) add README to marketplace and services repos explaining about the repository itself, how its versioning works, and the placeholders that can be used.

### CODE REVIEW

**Review by polillomm — COMMENTED** [review]
Dev review ok.

File: src/infra/envs/envs.go [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
-	AccessTokenCookieKey         string = "os-access-token"
-	UserDataBaseDirectory        string = "/home/"
+	InfiniteOsVersion               string = "0.1.5"
+	InfiniteOsMainDir               string = "/infinite"
+	InfiniteOsBinary                string = InfiniteOsMainDir + "/os"
+	InfiniteOsEnvFilePath           string = InfiniteOsMainDir + "/.env"
+	PersistentDatabaseFilePath      string = InfiniteOsMainDir + "/os.db"
+	TrailDatabaseFilePath           string = InfiniteOsMainDir + "/trail.db"
+	MarketplaceCatalogItemsDir      string = InfiniteOsMainDir + "/marketplace"
+	MarketplaceCatalogItemsVersion  string = "v1"
```
Comment (ntorga) [inline src/infra/envs/envs.go:11]: Branch, no?

File: src/infra/helper/dummyValueGenerator.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+func (helper *DummyValueGenerator) GenMailAddress(mailUsername *string) string {
+	dummyMailAddressDomains := []string{
+		"@enterprise.com", "@corporation.com", "@company.org",
+	}
+	randomMailAddressDomain := dummyMailAddressDomains[rand.Intn(len(dummyMailAddressDomains))]
+
+	if mailUsername == nil {
+		mailUsername = &helper.generatedUsername
```
Comment (ntorga) [inline src/infra/helper/dummyValueGenerator.go:40]: How do you know if generatedUsername isn't empty?

File: src/infra/marketplace/marketplaceCmdRepo.go [hunk 3]
```diff
@@ -753,10 +769,11 @@ func (repo *MarketplaceCmdRepo) RefreshCatalogItems() error {
 			return err
 		}
 
-		_, err = infraHelper.RunCmdWithSubShell(
-			"cd " + infraEnvs.InfiniteOsMainDir + ";" +
-				"git clone https://github.com/goinfinite/os-marketplace.git marketplace",
+		repoCloneCmd := fmt.Sprintf(
+			"cd %s; git clone --single-branch --branch %s https://github.com/goinfinite/os-marketplace marketplace",
```
Comment (ntorga) [inline src/infra/marketplace/marketplaceCmdRepo.go:773]: I'd have moved the URL of the repo to the infraEnvs.

File: src/infra/marketplace/marketplaceQueryRepo.go [hunk 4]
```diff
@@ -135,6 +135,24 @@ func (repo *MarketplaceQueryRepo) catalogItemMappingsFactory(
 	return itemMappings, nil
 }
 
+func (repo *MarketplaceQueryRepo) generateDataFieldDefaultValueBasedOnSpecificType(
```
Comment (ntorga) [inline src/infra/marketplace/marketplaceQueryRepo.go:138]: specificTypeDataFieldValueGenerator

File: src/infra/helper/dummyValueGenerator.go (hunk 2 above)
Comment (polillomm) [reply src/infra/helper/dummyValueGenerator.go:40]: I added a validation beforehand to ensure the value is filled in.

File: src/infra/envs/envs.go [hunk 5]
```diff
@@ -1,22 +1,24 @@
...
+	InfiniteOsBinary                string = InfiniteOsMainDir + "/os"
+	InfiniteOsEnvFilePath           string = InfiniteOsMainDir + "/.env"
+	PersistentDatabaseFilePath      string = InfiniteOsMainDir + "/os.db"
+	TrailDatabaseFilePath           string = InfiniteOsMainDir + "/trail.db"
+	MarketplaceCatalogItemsDir      string = InfiniteOsMainDir + "/marketplace"
+	MarketplaceCatalogItemsRepoUrl  string = "https://github.com/goinfinite/os-marketplace"
+	MarketplaceCatalogItemsBranch   string = "v1"
+	InstallableServicesItemsDir     string = InfiniteOsMainDir + "/services"
+	InstallableServicesItemsRepoUrl string = "https://github.com/goinfinite/os-services"
+	InstallableServicesItemsBranch  string = "v1"
```
Comment (ntorga) [inline src/infra/envs/envs.go:15]: If ItemsRepoUrl has the prefix Repo, so should the ItemsBranch. Just Branch isn't enough.

File: src/infra/helper/dummyValueGenerator.go [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+		"yoda", "obi_wan", "anakin", "luke", "leia", "rey", "kylo",
+	}
+	helper.generatedUsername = dummyUsernames[rand.Intn(len(dummyUsernames))]
+
+	return helper.generatedUsername
+}
+
+func (helper *DummyValueGenerator) GenMailAddress(mailUsername *string) string {
+	dummyMailAddressDomains := []string{
+		"@enterprise.com", "@corporation.com", "@company.org",
```
Comment (ntorga) [inline src/infra/helper/dummyValueGenerator.go:35]: If you're using SW chars, then this should be @republic.gov or @senate.gov just saying.

**Review by ntorga — CHANGES_REQUESTED** [review]

**Pull request merged**